packages/langium/src/node/node-file-system-provider.ts

```diff
@@ -0,0 +1,51 @@
+/******************************************************************************
+ * Copyright 2022 TypeFox GmbH
+ * This program and the accompanying materials are made available under the
+ * terms of the MIT License, which is available in the project root.
+ ******************************************************************************/
+
+import fs from 'fs';
+import { URI, Utils } from 'vscode-uri';
+import { FileSystemFilter, FileSystemNode, FileSystemProvider } from '../workspace/file-system-provider';
+
+export class NodeFileSystemProvider implements FileSystemProvider {
+
+    readFile(uri: URI): Promise<string> {
+        return fs.promises.readFile(uri.fsPath, 'utf-8');
```

Comment:
Proposal: make the encoding configurable by declaring a public property on this service class, initialized with `utf-8`.